### ISSUE TITLE
how can I use this model in  Kitti dataset

### ISSUE THREAD

**Issue by PleasantPhoenix** [CVMI-Lab/SparseKD#11]
Hello, how can I use this model in  Kitti dataset?I encountered the following error after changing the dataset.
![image](https://user-images.githubusercontent.com/86542147/230284019-522ea1f5-54cc-469d-97a2-224e8355a118.png)

**Comment by jihanyang** [comment]
I haven't support distillation with anchor head detector in this repo, maybe I will support it in the future update.

**Comment by PleasantPhoenix** [comment]
Can you please tell me how the distillation model based on second mentioned in your paper was implemented in the Kitti dataset?

**Comment by jihanyang** [comment]
Sure. The core is the PP logit KD and how we define the pivotal positions. we define the positive anchors when assigning target as the pivotal positions and carry on logit distillation on these anchors. The ditillation of classification head uses L2 loss with loss weight 15.0, and the distillation of regression head uses smooth L1 loss with loss weight 0.2.

**Comment by charmeleonz** [comment]
> 

Hi there, just want to confirm with you that throughout the paper, the PP logit KD is only applied to the cls part of output KD, rather than the reg part right？

**Comment by jihanyang** [comment]
@charmeleonz  I applied it on the regression part by default unless there are input resolution gap (voxel size difference) between the teacher and student. You can also check our configs for cp-voxel (use it) and cp-pillar (not use) for that thing.

**Comment by charmeleonz** [comment]
> @charmeleonz I applied it on the regression part by default unless there are input resolution gap (voxel size difference) between the teacher and student. You can also check our configs for cp-voxel (use it) and cp-pillar (not use) for that thing.

Thanks for the reply. I may not have made it very clear. I was wondering whether the soft/fg mask computed is only applied to the 'classification' loss calculation, but not the 'localisation' loss?

**Comment by jihanyang** [comment]
Compare the following config and re-read my previous clarificaiton might help you understand: 
For model compression, the pp logit kd loss weight for regression is 0.2 :
https://github.com/CVMI-Lab/SparseKD/blob/8572093edc77d4217f7596d9e1de68c6dc77e420/tools/cfgs/waymo_models/cp-voxel/cp-voxel-s_sparsekd.yaml#L302-L307
and input compression,  the pp logit kd loss weight for regression is 0 :
https://github.com/CVMI-Lab/SparseKD/blob/8572093edc77d4217f7596d9e1de68c6dc77e420/tools/cfgs/waymo_models/cp-pillar/cp-pillar-v0.48_sparsekd.yaml#L307-L312

**Comment by charmeleonz** [comment]
@jihanyang Hi, thanks for the info. My confusion was: 
Regression part = classification + localisation.
But the attention mask is never ever applied to  'localisation' -> is this statement true?

**Comment by jihanyang** [comment]
In openpcdet, regression here means your 'localisation'. If you have checked the config, you will see `classification loss` and `regression_loss`, which might address your confuse.

**Comment by carlvinson70** [comment]
@jihanyang Interesting. Is any of the masks named 'soft_mask', 'fg_mask' or 'rank_mask', ever applied to the "regression_loss", or let's say the 'kd_reg_loss'? Cos I only saw it being applied to the cls part, or the 'kd_hd_loss'
